app/src/main/java/in/ac/bits_pilani/goa/ard/interfaces/NavigationDrawerListener.java

```diff
@@ -0,0 +1,133 @@
+package in.ac.bits_pilani.goa.ard.interfaces;
+
+import android.content.Context;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
+import com.bumptech.glide.request.RequestOptions;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Random;
+
+import in.ac.bits_pilani.goa.ard.R;
+import in.ac.bits_pilani.goa.ard.activities.MainActivity;
+import in.ac.bits_pilani.goa.ard.utils.AHC;
+
+/**
+ * NavigationDrawerListener Class.
+ * reflects changes from firebase regarding nav drawer header
+ *
+ * @author Rushikesh Jogdand
+ */
+
+public class NavigationDrawerListener implements ValueEventListener {
+
+    /**
+     * Context required for glide image loading.
+     */
+    private final Context context;
+
+    /**
+     * Textview corresponding to nav drawer title.
+     */
+    private final TextView navDrawerTitle;
+
+    /**
+     * Textview corresponding to nav drawer subtitle.
+     */
+    private final TextView navDrawerSubtitle;
+
+    /**
+     * ImageView corresponding to nav drawer header background.
+     */
+    private final ImageView navDrawerImage;
+
+    /**
+     * TAG for methods in this class.
+     */
+    private final String TAG;
```

Comment:
add TAG value here itself instead of passing it as a parameter in the constructor